hscontrol/mapper/mapper.go

```diff
@@ -273,13 +270,21 @@ func (m *Mapper) PeerChangedResponse(
 	}
 
 	changedNodes := make(types.Nodes, 0, len(changedIDs))
-	for _, peer := range peers {
-		if slices.Contains(changedIDs, peer.ID) {
-			changedNodes = append(changedNodes, peer)
+	for _, changedID := range changedIDs {
+		if changedID != node.ID {
+			changedNode, err := m.GetNode(changedID)
```

Comment:
The problem with doing it this way, and not as mentioned above is that we dont get transactional consistency. In this case, nodes might change in between calls to GetNode. While this might not be an obvious problem, it might lead to incredibly hard to debug problems.

Reply:
I have reworked the code to use exactly one database query. I will also add a test to the database query function at least.